tools/apigee-envoy-quickstart/extension/README.md

```diff
@@ -0,0 +1,156 @@
+    ```bash
+    export TARGET_HOST="httpbin.org"
+    export TARGET_SERVICE_NAME="httpbin"
+    export TARGET_SERVICE_NAMESPACE="apigee"
+    ```
+
+1. **Fetch the Loadbalancer details exposing the ingress gatway**
+    ```bash
+    export INGRESS_HOST=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    export INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+    export SECURE_INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+    export TCP_INGRESS_PORT=$(kubectl -n $ISTIO_GATEWAY_NS get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="tcp")].port}')
+    ```
+
+1. **Setup Gateway component**
+    ```bash
+   cat <<EOF > /tmp/httpbin-gateway.tmpl              
+   apiVersion: networking.istio.io/v1alpha3
+   kind: Gateway
+   metadata:
+     name: httpbin-gateway
+   spec:
+     selector:
+       istio: ingressgateway
+     servers:
+       - port:
+           number: 80
+           name: http
+           protocol: HTTP
+         hosts:
+           - "$TARGET_HOST"
+   EOF
+
+   envsubst < /tmp/httpbin-gateway.tmpl > /tmp/httpbin-gateway.yaml
+
+   kubectl apply -n $ISTIO_GATEWAY_NS -f /tmp/httpbin-gateway.yaml
+    ```
+
+1. **Configure routes for traffic entering via the Gateway:**
+    ```bash
+   cat <<EOF > /tmp/httpbin-virtual-service.tmpl
+   apiVersion: networking.istio.io/v1alpha3
+   kind: VirtualService
+   metadata:
+     name: httpbin-ingress
+   spec:
+     hosts:
+     - "$TARGET_HOST"
+     gateways:
+     - httpbin-gateway
+     http:
+     - route:
+       - destination:
+           host: $TARGET_SERVICE_NAME.$TARGET_SERVICE_NAMESPACE.svc.cluster.local
+           port:
+             number: 80
+   EOF
+
+   envsubst < /tmp/httpbin-virtual-service.tmpl > /tmp/httpbin-virtual-service.yaml
+    
+   kubectl apply -n $ISTIO_GATEWAY_NS -f /tmp/httpbin-virtual-service.yaml
+    ```
+
+1. **Extract the consumer key from the developer app, this is created on Apigee platform during the [quickstart setup](https://github.com/apigee/devrel/tree/main/tools/apigee-envoy-quickstart#envoy-with-apigee-adapter-as-containers-within-kubernetes-platform) :**
+    ```bash
+    CONSUMER_KEY=$(curl -H "Authorization: Bearer ${TOKEN}"  \
+    -H "Content-Type:application/json" \
+    "https://apigee.googleapis.com/v1/organizations/${APIGEE_ORG}/developers/test-envoy@google.com/apps/envoy-adapter-app-2" | \
+    jq '.credentials[0].consumerKey'); \
+    CONSUMER_KEY=$(echo "$CONSUMER_KEY"|cut -d '"' -f 2); 
+    ```
+
+1. **Extract IP address exposed via ingressgateway component**
+    ```bash
+    EXTERNAL_IP=$(kubectl --context="${CLUSTER_CTX}" get svc -n "$ISTIO_GATEWAY_NS" -o json | \
```

Comment:
don't you have this already from above `INGRESS_HOST`

Reply:
yepp... cleaning it